Doc - format for number and dates

Cuervino opened the pull request:
**Component**:

MDX documentation on Format
Style>Format

**Description**:
Add 1 page with a guideline about number format and date format, in order to have consistency in the UI

**Design**:
![image](https://user-images.githubusercontent.com/28654802/112843523-f4bc1280-90a2-11eb-8463-e63c8ed2138b.png)
![image](https://user-images.githubusercontent.com/28654802/112843871-5e3c2100-90a3-11eb-8155-b40a73bbcaca.png)


**Breaking Changes**:
no
- [] Breaking Changes


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

Cuervino committed:
Add 1 page with guideline about number format and date format, in order to have consistency in the UI

bert-e commented:
# Hello cuervino,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/core-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @monpote

* @nicolas2bert

* @jbwatenbergscality

* @rachedbenmustapha

* @carlito-scality

* @chengyanjin

JBWatenbergScality commented on stories/format.stories.mdx:
```diff
@@ -0,0 +1,44 @@
+| ------- | ------------------------ | ------------------------ | ------ | ---------------------------------------------------- |
+| Short#1 | DD MMM                   | 20 Jul                   | 5      | Chart time axis                                      |
+| Short#2 | DDMMM HH:mm              | 20JUL 09:00              | 11     | Limited space - Dense tables, charts                 |
+| Short#3 | YYYY-MM-DD               | 2020-07-20               | 10     |                                                      |
+| Mid#1   | EEE YYYY-MM-DD HH:mm     | Mon 2020-07-20 09:00     | 20     | Checking with agenda for day of the week             |
+| Mid#2   | YYYY-MM-DD HH:mm         | 2020-07-20 09:00         | 16     | No need of day of the week                           |
+| Mid#3   | YYYY-MM-DD HH:mm:ss      | 2020-07-20 09:00:00      | 19     | When the seconds are needed                          |
+| Full#1  | EEE MMM DD YYYY HH:mm:ss | Mon Jul 20 2020 09:00:00 | 24     | When a lot of space (hover) When precision is needed |
```
Shouldn't this be internationalised and for example use `MM/DD/YYYY` in the US, and `DD/MM/YYYY` in Europe ?

Cuervino replied:
Yes, from the long-term point of view. For that, we would need to add more "clues" about whether we are displaying US format or non-US format.
Having an English language and just for example "12/11/2021" is not enough for knowing if it's December or November of 2021. So we might need to add more information (letters, like 12NOV2021, etc.).
To avoid ambiguity, the easier and most straight forward way would be to use the YYYY-MM-DD format for the moment.

thomasdanan commented on stories/format.stories.mdx:
```diff
@@ -0,0 +1,44 @@
+| Short#1 | DD MMM                   | 20 Jul                   | 5      | Chart time axis                                      |
+| Short#2 | DDMMM HH:mm              | 20JUL 09:00              | 11     | Limited space - Dense tables, charts                 |
+| Short#3 | YYYY-MM-DD               | 2020-07-20               | 10     |                                                      |
+| Mid#1   | EEE YYYY-MM-DD HH:mm     | Mon 2020-07-20 09:00     | 20     | Checking with agenda for day of the week             |
```
Do we really need Mid#1, not sure where we are using it currently?
I guess we should try to reduce this list as much as possible ...

Cuervino replied:
I still think that it makes sense to have the standard date with 2 formats: with and without seconds.
Space can be very scarce.

thomasdanan commented on stories/format.stories.mdx:
```diff
@@ -0,0 +1,44 @@
+| Alias   | code                     | Example                  | Length | Context                                              |
+| ------- | ------------------------ | ------------------------ | ------ | ---------------------------------------------------- |
+| Short#1 | DD MMM                   | 20 Jul                   | 5      | Chart time axis                                      |
+| Short#2 | DDMMM HH:mm              | 20JUL 09:00              | 11     | Limited space - Dense tables, charts                 |
```
Is it really the one we want to use in dense tables? I was thinking the MMM was causing some alignment issues and that it was preferable to use numeric format?

Cuervino replied:
Correct, I change the "table" use cases.

thomasdanan commented on stories/format.stories.mdx:
```diff
@@ -0,0 +1,44 @@
+
+### Format list
+
+| Alias   | code                     | Example                  | Length | Context                                              |
```
more generally, I tend to think we should limit as much as possible the number of formats we use in the UIs ...

Cuervino replied:
Correct, I reduced the list to 6.

ChengYanJin reviewed (commented):
I feel it would interesting for me to some examples, meaning in which case I should use this format.

Cuervino committed:
update table

2 formats removed from the list
table formats with no letters
MMM always in Pascal Case

Cuervino commented:
> 
> 
> I feel it would interesting for me to some examples, meaning in which case I should use this format.
What do you mean exactly?
You mean associated screens? Visual exemples? It may be in a next version of this page.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolas2bert

* @carlito-scality

* @monpote

* @chengyanjin

* @jbwatenbergscality

* @rachedbenmustapha

Cuervino commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue None.

Goodbye cuervino.
